lib/fn.js

```diff
@@ -9,34 +12,49 @@
  */
 export function debounce(fn, timeout) {
 
-  var timer;
+  let timer;
 
-  var lastArgs;
-  var lastThis;
+  let lastArgs;
+  let lastThis;
 
-  var lastNow;
+  let lastNow;
 
-  function fire() {
+  function fire(force) {
 
-    var now = Date.now();
+    let now = Date.now();
 
-    var scheduledDiff = (lastNow + timeout) - now;
+    let scheduledDiff = force ? 0 : (lastNow + timeout) - now;
 
     if (scheduledDiff > 0) {
       return schedule(scheduledDiff);
     }
 
     fn.apply(lastThis, lastArgs);
 
-    timer = lastNow = lastArgs = lastThis = undefined;
+    clear();
   }
 
   function schedule(timeout) {
     timer = setTimeout(fire, timeout);
   }
 
-  return function(...args) {
+  function clear() {
+    if (timer) {
+      clearTimeout(timer);
+    }
 
+    timer = lastNow = lastArgs = lastThis = undefined;
+  }
+
+  function flush() {
+    if (timer) {
+      fire(true);
+    }
+
+    clear();
```

Comment:
We are calling `clear()` [twice if timer is set](https://github.com/bpmn-io/min-dash/pull/20/files#diff-1b3f81b82a473a44ed3c5f5e16fce0030abc3b0cdf5efac082ba1fc01a9de490R34)

Reply:
I could also go for this one here: 

```
  function flush() {
    if (timer) {
      fire(true);
    } else {
      clear();
    }
  }
```

But would that not make this piece harder to reason about?

Reply:
Obviously we could call the method `fireAndClear` but then... it is only internals.

Reply:
ok let's leave it as it is